cibuildwheel/__main__.py

```diff
@@ -164,7 +164,27 @@ def print_preamble(platform, build_options):
     for option, value in build_options.items():
         print('  %s: %r' % (option, value))
 
+    warnings = detect_warnings(platform, build_options)
+    if warnings:
+        print('\nWarnings:')
+        for warning in warnings:
+            print('  ' + warning)
+
     print('\nHere we go!\n')
 
+def detect_warnings(platform, build_options):
+    warnings = []
+
+    # warn about deprecated {python} and {pip}
+    for option_name in ['test_command', 'before_build']:
+        option_value = build_options.get(option_name)
+
+        if option_value:
```

Comment:
For the sake of being slightly shorter and non-nested: `build_options.get(option_name, '')` and remove this if statement?

Reply:
good idea

Reply:
Aaaaaah, dang. Well, then the only thing to stop nesting is to combine the if statements, but ... bleh, not worth a commit :-(